### DISCUSSION TITLE
Regarding move dialog of pcmanfm-qt

### DISCUSSION THREAD

**Discussion by Lokawn** [lxqt/lxqt#2259] [Ideas]
![screengrab-2022-06-24-14-27-58](https://user-images.githubusercontent.com/39772448/176246624-bacded5e-702f-4a1b-955e-d5283691bc23.png)
The move dialog of shows change in files processed only when an individual file finishes moving and not in realtime, so wouldn't it be more correct to show the number of files moved instead of the amount of data moved?

**Comment by tsujan** [comment]
> so wouldn't it be more correct to show the number of files moved instead of the amount of data moved?

The number of files isn't informative in the case of file transference. Moreover, it would make no difference in the time of showing the info (read below).

Your screenshot usually happens when moving files between different filesystem IDs (`FILE_ATTRIBUTE_ID_FILESYSTEM`), in which case, the files are first copied to the destination, and if successful, they are deleted from the source.

Now, when the copying is started, it's announced by the progressbar, but only when copying is finished, the size of the processed file is announced. The reason is that we don't know beforehand that the file will be copied (as a result of an overwrite prompt, lack of permission, etc.).

When the copied file is a folder, we have to enumerate its contents to transfer them. A folder can contain other folders, each of which may contain other folders, etc. The transference time is out of our control for other reasons too — drive speed, possible prompts, permission errors, network speed (e,g., when transferring files to Google drive), etc.
